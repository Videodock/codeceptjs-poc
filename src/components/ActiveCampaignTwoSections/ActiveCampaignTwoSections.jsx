import React from 'react';
import { makeStyles } from '@material-ui/styles';
import { Typography } from '@material-ui/core';
import classNames from 'classnames';
import * as PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import OpenInNewIcon from '@material-ui/icons/OpenInNew';
import images from '../../theme/images';
import styles from './ActiveCampaignTwoSections.styles';

const useStyles = makeStyles(styles);

export default function ActiveCampaignTwoSections ({ campaign, digitalDocument }) {
  const classes = useStyles();

  return (
    <Grid
      direction="row"
      justify="center"
      alignItems="center"
      container
    >
      <Grid className={classNames(classes.twoSections, classes.pieceInfo)} xs={12} md={6} item>
        <section>
          <Typography variant="h2">
            Mahler: Symphony No. 6 in A minor
          </Typography>
          <Typography gutterBottom>
            Mahler’s A-minor Sixth Symphony is a mythical piece. Mahler may or may not have subtitled it “Tragic” at some stage of its composition, and it could, possibly, contain music that consecrates and depicts his wife Alma. It may be “the first nihilist work in the history of music”, as conductor Wilhelm Furtwängler described it. Conductor and friend of Mahler’s Bruno Walter found the piece too expressively dark for him to conduct, since it “ends in hopelessness and the dark night of the soul”. Most significantly, it’s a work you are always told is dangerously, prophetically autobiographical, above all in its final fourth movement, that half-hour-long hallucinogenic, emotional nightmare-scape.
          </Typography>
          <Typography variant="h2">
          Complete Score
          </Typography>
          <Typography gutterBottom>
            #514284 - 45.31MB - 263 pages - 486× downloaded
            PDF scanned by US-AAu
            Music Addict (2018/3/2)
          </Typography>
          <a className={classes.moreLink} target="_blank" rel="noopener noreferrer" href={digitalDocument?.source}>
            <OpenInNewIcon />
            More
          </a>
        </section>
      </Grid>
      <Grid className={classNames(classes.twoSections, classes.pieceAuthor)} xs={12} md={6} item>
        <section>
          <img className={classes.mahlerHeadshot} src={images.mahlerHeadshot} alt="" />
          <Typography variant="h2">
          Gustav Mahler
          </Typography>
          <Typography gutterBottom>
          (7 July 1860 — 18 May 1911)
          </Typography>
          <Typography gutterBottom>
          Gustav Mahler was an Austro-Bohemian Romantic composer, and one of the leading conductors of his generation. As a composer he acted as a bridge between the 19th century Austro-German tradition and the modernism of the early 20th century. While in his lifetime his status as a conductor was established beyond question, his own music gained wide popularity only after periods of relative neglect, which included a ban on its performance in much of Europe during the Nazi era. After 1945 his compositions were rediscovered by a new generation of listeners; Mahler then became one of the most frequently performed and recorded of all composers, a position he has sustained into the 21st century.
          </Typography>
        </section>
      </Grid>
    </Grid>
  );
}

ActiveCampaignTwoSections.propTypes = {
  campaign       : PropTypes.object,
  digitalDocument: PropTypes.object,
};

ActiveCampaignTwoSections.defaultProps = {};
