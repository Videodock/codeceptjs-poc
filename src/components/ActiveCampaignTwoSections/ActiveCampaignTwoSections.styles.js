import { createStyles } from '@material-ui/styles';

export default ({ spacing, typography, palette, breakpoints }) => createStyles({
  relative: {
    position: 'relative',
  },
  twoSections: {
    display                 : 'flex',
    height                  : 600,
    maxHeight               : 600,
    [breakpoints.only('sm')]: {
      height: 500,
    },
    '& section': {
      width : 528,
      margin: '51px 96px 98px 96px',
      '& h2': {
        marginBottom: spacing(0.5),
      },
      '& p': {
        marginBottom: spacing(2),
      },
      [breakpoints.down('md')]: {
        marginTop   : spacing(4),
        marginLeft  : spacing(2),
        marginRight : spacing(2),
        marginBottom: spacing(4),
      },
      [breakpoints.only('sm')]: {
        marginLeft  : spacing(4),
        marginRight : spacing(4),
        marginBottom: spacing(4),
      },
      [breakpoints.only('xs')]: {
        marginLeft : spacing(2),
        marginRight: spacing(2),
      },
    },
  },
  pieceInfo: {
    justifyContent          : 'flex-end',
    [breakpoints.down('sm')]: {
      justifyContent: 'flex-start',
    },
  },
  pieceAuthor: {
    justifyContent          : 'flex-start',
    backgroundColor         : palette.common.faintGrey,
    [breakpoints.only('xs')]: {
      marginBottom: spacing(-3),
    },
    '& section': {
      display       : 'flex',
      flexDirection : 'column',
      justifyContent: 'center',
      alignItems    : 'center',
    },
  },
  mahlerHeadshot: {
    borderRadius: '50%',
    width       : 115,
    height      : 115,
  },
});
