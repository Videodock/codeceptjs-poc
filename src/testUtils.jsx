import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
