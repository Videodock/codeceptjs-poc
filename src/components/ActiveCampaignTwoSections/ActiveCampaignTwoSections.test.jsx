import React from 'react';
import { render } from '../../testUtils';
import ActiveCampaignTwoSections from './ActiveCampaignTwoSections';

describe('<ActiveCampaignTwoSections />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<ActiveCampaignTwoSections />);

    expect(container.firstChild).toMatchSnapshot();
  });

  // add your unit tests here
});
