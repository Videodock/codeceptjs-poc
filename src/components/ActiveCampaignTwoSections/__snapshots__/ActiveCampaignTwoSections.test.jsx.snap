// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActiveCampaignTwoSections /> renders and matches snapshot 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    class="MuiGrid-root makeStyles-twoSections-2 makeStyles-pieceInfo-3 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
  >
    <section>
      <h2
        class="MuiTypography-root MuiTypography-h2"
      >
        Mahler: Symphony No. 6 in A minor
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        Mahler’s A-minor Sixth Symphony is a mythical piece. Mahler may or may not have subtitled it “Tragic” at some stage of its composition, and it could, possibly, contain music that consecrates and depicts his wife Alma. It may be “the first nihilist work in the history of music”, as conductor Wilhelm Furtwängler described it. Conductor and friend of Mahler’s Bruno Walter found the piece too expressively dark for him to conduct, since it “ends in hopelessness and the dark night of the soul”. Most significantly, it’s a work you are always told is dangerously, prophetically autobiographical, above all in its final fourth movement, that half-hour-long hallucinogenic, emotional nightmare-scape.
      </p>
      <h2
        class="MuiTypography-root MuiTypography-h2"
      >
        Complete Score
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        #514284 - 45.31MB - 263 pages - 486× downloaded PDF scanned by US-AAu Music Addict (2018/3/2)
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
              />
            </svg>
          </span>
          More
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </section>
  </div>
  <div
    class="MuiGrid-root makeStyles-twoSections-2 makeStyles-pieceAuthor-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
  >
    <section>
      <img
        alt=""
        class="makeStyles-mahlerHeadshot-5"
        src="mahler_headshot.jpg"
      />
      <h2
        class="MuiTypography-root MuiTypography-h2"
      >
        Gustav Mahler
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        (7 July 1860 — 18 May 1911)
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        Gustav Mahler was een in Bohemen geboren en opgegroeide Oostenrijkse componist en dirigent van joodse afkomst. Mahler gold als een van de belangrijkste dirigenten van zijn tijd, maar wordt tegenwoordig vooral gezien als componist die de late romantiek verbonden heeft met de moderne periode van de klassieke muziek die met name in de Tweede Weense School gestalte kreeg. Als dirigent was hij onder andere actief aan de operahuizen van Boedapest en Hamburg en aan de hofopera te Wenen. In Wenen was hij tevens dirigent van de Wiener Philharmoniker.
      </p>
    </section>
  </div>
</div>
`;
